src/curve.rs

```diff
@@ -672,12 +525,62 @@ impl_binops_multiplicative_mixed!(AffinePoint, Scalar, ProjectivePoint);
 
 #[inline(always)]
 const fn mul_by_3b(a: &Fp6) -> Fp6 {
-    a.mul(&B3)
+    let aa = (&a.c0).mul(&B3.c0).0 as u128;
+    let ab = a.c0.0 as u128;
+    let ab = ab + (ab << 1);
+
+    let ba = (&a.c1).mul(&B3.c0).0 as u128;
+    let bb = a.c1.0 as u128;
+    let bb = bb + (bb << 1);
+
+    let ca = (&a.c2).mul(&B3.c0).0 as u128;
+    let cb = a.c2.0 as u128;
+    let cb = cb + (cb << 1);
+
+    let da = (&a.c3).mul(&B3.c0).0 as u128;
+    let db = a.c3.0 as u128;
+    let db = db + (db << 1);
+
+    let ea = (&a.c4).mul(&B3.c0).0 as u128;
+    let eb = a.c4.0 as u128;
+    let eb = eb + (eb << 1);
+
+    let fa = (&a.c5).mul(&B3.c0).0 as u128;
+    let fb = a.c5.0 as u128;
+    let fb = fb + (fb << 1);
+
+    let c0 = fb * GENERATOR.0 as u128;
+    let c0 = c0 + aa;
+    let c0 = Fp(reduce_u96(c0));
+
+    let c1 = ab + ba;
```

Comment:
Just my OCD: could it be `ba + ab` for consistency with lower lines? Could also have `let c0= aa + c0;` above, but that already breaks the pattern.

Reply:
That's funny, my OCD logic worked differently there: ordering the ab/ac/ba... alphabetically but when adding something to an existing non-mutable variable I always write the overwritten one on the left. No hard feelings though, so I can change it if you want :)

Reply:
But then doesn't bb come before ca? Anyway, not really an issue. As long as someone's OCD is satisfied :P

Reply:
Ah I hadn't seen below, fixed this indeed :p